scripts/eval/eval.py

```diff
@@ -1,76 +1,238 @@
 # Copyright 2022 MosaicML LLM Foundry authors
 # SPDX-License-Identifier: Apache-2.0
 
+import re
 import sys
 import time
 from typing import List
 
+import pandas as pd
 import torch
 from composer.loggers import InMemoryLogger, LoggerDestination
 from composer.trainer import Trainer
 from composer.utils import dist, get_device, reproducibility
 from omegaconf import DictConfig
 from omegaconf import OmegaConf as om
 
+from llmfoundry.callbacks import ModelGauntlet
 from llmfoundry.models.model_registry import COMPOSER_MODEL_REGISTRY
 from llmfoundry.utils.builders import (build_icl_evaluators, build_logger,
                                        build_tokenizer)
 
 
+def load_model(model_cfg, tokenizer):
+    retries = 0
+    while retries < 3:
+        try:
+            composer_model = COMPOSER_MODEL_REGISTRY[model_cfg.name](model_cfg,
+                                                                     tokenizer)
+            return composer_model
+        except Exception as e:
+            retries += 1
+            if retries >= 3:
+                raise e
+            else:
+                print(
+                    f'Got exception {str(e)} while loading model {model_cfg.name}. {3-retries} retries remaining'
+                )
+    return None
```

Comment:
is this needed? i think python just returns None if the fn does not have a return.